src/Mjolnir.ts

```diff
@@ -951,18 +951,22 @@ export class Mjolnir {
     }
 
     private async handleConsequence(protection: Protection, roomId: string, eventId: string, sender: string, consequence: Consequence) {
-        switch (consequence.type) {
-            case ConsequenceType.alert:
-                break;
-            case ConsequenceType.redact:
-                await this.client.redactEvent(roomId, eventId, "abuse detected");
-                break;
-            case ConsequenceType.ban:
-                await this.client.banUser(sender, roomId, "abuse detected");
-                break;
+        let met: string[] = [];
+
+        if (consequence.alert) {
+            met.push("alert");
+        }
+        if (consequence.ban) {
+            met.push("ban");
+            await this.client.banUser(sender, roomId, "abuse detected");
+        }
+        if (consequence.redact) {
+            met.push("redact");
+            await this.client.redactEvent(roomId, eventId, "abuse detected");
```

Comment:
So this is getting a little messy. 

The problem is the `Consequence` instance is describing the outcome of a protection which now is really multiple consequences and it's having to describe all of them at the same time in one place. Meaning this `handleConsequence` method will get big and troublesome to edit. 

Instead we could have something like an `handleProtectionOutcome` and an `Outcome` class which then just has an array with a bunch of consequences in (e.g. ban + redact) (and maybe some summary information). Then the consequences can then have methods on them to describe what it does and a method to do the consequence. Then we're not needing to keep adding more to this single method bc it's it'll just be calling code.

Reply:
i opted for multiple actions in the same `Consequence` so there's 1 reason for each action, but that's got upsides as well as downsides. maybe it could just handle a `Consequence[]` rather than a new thing like `Outcome`

Reply:
Yeah, that might work